### ISSUE TITLE
Pulumi waits for 300 seconds for healthz in case of cluster with only private endpoint enabled

### ISSUE THREAD

**Issue by RaviVadera** [pulumi/pulumi-eks#1133]
### What happened?

When creating a ```eks.Cluster``` with only private endpoint enabled, cluster healthz check waits for 300 seconds, knowing that it won't be able to perform the healthz check

**What happened**
Pulumi warns,
Waiting for cluster to be ready, retries goes all the way to 0 seconds
```
@ updating....
    aws:eks:Cluster xxx-cluster-eksCluster  Waiting up to (300) more seconds for cluster readiness...
```

**Expected to happen**
Pulumi shold not try to perform healthz check for the cluster if only private endpoint is enabled.

### Example

Create a cluster with following params,
```
endpointPublicAccess: false,
endpointPrivateAccess: true,
```

### Output of `pulumi about`

```
CLI
Version      3.113.3
Go Version   go1.22.2
Go Compiler  gc

Plugins
KIND      NAME        VERSION
resource  aws         6.31.0
resource  awsx        2.7.0
resource  docker      4.5.3
resource  docker      3.6.1
resource  eks         2.3.0
resource  kubernetes  4.10.0
language  nodejs      unknown
resource  random      4.16.1

Host
OS       ubuntu
Version  22.04
Arch     x86_64

This project is written in nodejs: executable='/usr/bin/node' version='v18.20.0'

Current Stack: organization/xxx/xxx

xxxxxxxx

Found no pending operations associated with dev

Backend
Name           xxxxx
URL            s3://xxxx
User           xxxx
Organizations
Token type     personal

Dependencies:
NAME                              VERSION
@typescript-eslint/eslint-plugin  7.7.0
eslint                            8.57.0
@pulumi/eks                       2.3.0
@pulumi/pulumi                    3.113.0
@pulumi/kubernetes                4.10.0
proxy-agent                       6.4.0
@types/node                       20.12.3
@typescript-eslint/parser         7.7.0
eslint-config-prettier            9.1.0
eslint-plugin-prettier            5.1.3
prettier                          3.2.5
@pulumi/awsx                      2.7.0
@pulumi/random                    4.16.1
@pulumi/aws                       6.31.0
yaml                              2.4.1

Pulumi locates its logs in /tmp by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by rquitales** [comment]
Thanks for reporting this issue. I agree, we should not attempt to communicate with the api server if the cluster only has private access!

**Comment by RaviVadera** [comment]
Proposed fix in #1135
